packages/dev/core/src/Rendering/fluidRenderer/fluidRenderer.ts

```diff
@@ -0,0 +1,547 @@
+     * Rebuilds the elements related to this component in case of
+     * context lost for instance.
+     */
+    public rebuild(): void {
+        if (this.scene._fluidRenderer) {
+            // Release resources first
+            this.scene.disableFluidRenderer();
+
+            // Re-enable
+            this.scene.enableFluidRenderer();
+        }
+    }
+
+    /**
+     * Disposes the component and the associated resources
+     */
+    public dispose(): void {
+        this.scene.disableFluidRenderer();
+    }
+}
+
+/**
+ * An object rendered as a fluid.
+ * It consists of the object itself as well as the render target renderer (which is used to generate the textures (render target) needed for fluid rendering)
+ */
+export interface IFluidRenderingRenderObject {
+    /** object rendered as a fluid */
+    object: FluidRenderingObject;
+    /** target renderer used to render the fluid object */
+    targetRenderer: FluidRenderingTargetRenderer;
+}
+
+/**
+ * Class responsible for fluid rendering.
+ * It is implementing the method described in https://developer.download.nvidia.com/presentations/2010/gdc/Direct3D_Effects.pdf
+ */
+export class FluidRenderer {
+    /** @internal */
+    public static _SceneComponentInitialization(scene: Scene) {
+        let component = scene._getComponent(SceneComponentConstants.NAME_FLUIDRENDERER) as FluidRendererSceneComponent;
+        if (!component) {
+            component = new FluidRendererSceneComponent(scene);
+            scene._addComponent(component);
+        }
+    }
+
+    private _scene: Scene;
+    private _engine: Engine;
+    private _onEngineResizeObserver: Nullable<Observer<Engine>>;
+    private _renderObjects: Array<IFluidRenderingRenderObject>;
+    private _targetRenderers: FluidRenderingTargetRenderer[];
+    private _cameras: Map<Camera, [Array<FluidRenderingTargetRenderer>, { [key: string]: FluidRenderingDepthTextureCopy }]>;
+
+    /** Retrieves all the render objects managed by the class */
+    public get renderObjects() {
+        return this._renderObjects;
+    }
+
+    /** Retrieves all the render target renderers managed by the class */
+    public get targetRenderers() {
+        return this._targetRenderers;
+    }
+
+    /**
+     * Initializes the class
+     * @param scene Scene in which the objects are part of
+     */
+    constructor(scene: Scene) {
+        this._scene = scene;
+        this._engine = scene.getEngine();
+        this._onEngineResizeObserver = null;
+        this._renderObjects = [];
+        this._targetRenderers = [];
+        this._cameras = new Map();
+
+        FluidRenderer._SceneComponentInitialization(this._scene);
+
+        this._onEngineResizeObserver = this._engine.onResizeObservable.add(() => {
+            this._initialize();
+        });
+    }
+
+    /**
+     * Reinitializes the class
+     * Can be used if you change the object priority (FluidRenderingObject.priority), to make sure the objects are rendered in the right order

```

Comment:
I didn't want to call it automatically because:
* it calls `_initialize()` which recreates all the textures / RTTs, so fairly heavy processing, and if you update the `priority` property for multiple objects in sequence, you will trigger the call for each change instead of calling `recreate` manually once at the end of the changes
* tracking changes to the `priority` property will require setting up an observable that the `FluidRenderer` will need to register with to be informed of any changes
* changing the priorities will probably be done in an initialization step most of the time and before any rendering has been done. In this case you, don't even need to call `recreate()` since everything will be created correctly the first time you use the fluid renderer.

But if you still think we should automatically call `recreate` when a `priority` changes, I will make the changes.